File: pizza_delivery_java/src/main/java/br/com/academiadoflutter/pizzaDeliveryApi/services/OrderService.java
package br.com.academiadoflutter.pizzaDeliveryApi.services;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.stereotype.Service;

import br.com.academiadoflutter.pizzaDeliveryApi.entities.MenuItemE;
import br.com.academiadoflutter.pizzaDeliveryApi.entities.OrderE;
import br.com.academiadoflutter.pizzaDeliveryApi.entities.OrderItemE;
import br.com.academiadoflutter.pizzaDeliveryApi.entities.UserE;
import br.com.academiadoflutter.pizzaDeliveryApi.repositories.OrderRepository;
import br.com.academiadoflutter.pizzaDeliveryApi.view_models.SaveOrderInputModel;
import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class OrderService {
  private final OrderRepository orderRepository;

  public List<OrderE> findAll() {
    return orderRepository.findAll();
  }

  public void saveOrder(SaveOrderInputModel saveModel) {
    
    var order = new OrderE();
    var user = new UserE();
    user.setId(saveModel.getUserId());

    order.setUser(user);
    order.setAddress(saveModel.getAddress());
    order.setPaymentType(saveModel.getPaymentType());
    order.setItems(saveModel.getItemsId().stream().map((i) -> {
      final var menu = new MenuItemE();
      menu.setId(i);
      final var item = new OrderItemE();
      item.setOrder(order);
      item.setMenu(menu);
      return item;
    }).collect(Collectors.toList()));
    orderRepository.save(order);
  }

  public List<OrderE> findMyOrders(Long userId) {
    return orderRepository.findByUserIdOrderById(userId);
  }
}

File: pizza_delivery_dart/lib/services/order/order_service.dart
import 'package:pizza_delivery/entities/order.dart';
import 'package:pizza_delivery/modules/order/view_model/save_order_input_model.dart';
import 'package:pizza_delivery/repositories/order/i_order_repository.dart';
import 'package:pizza_delivery/services/order/i_order_service.dart';
import 'package:injectable/injectable.dart';

@LazySingleton(as: IOrderService)
class OrderService implements IOrderService {
  final IOrderRepository _repository;

  OrderService(this._repository);

  @override
  Future<void> saveOrder(SaveOrderInputModel saveOrder) async {
    await _repository.saveOrder(saveOrder);
  }

  @override
  Future<List<Order>> findOrder(int userId) async {
    return await _repository.findOrder(userId);
  }
}
